kie-server-parent/kie-server-services/kie-server-services-jbpm-queries/src/main/java/org/kie/server/services/jbpm/queries/util/QueryStrategy.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.services.jbpm.queries.util;
+
+import java.util.Map;
+
+import org.jbpm.services.api.query.model.QueryParam;
+
+/**
+ * Provides DB specific query and column-mappings for queries.
+ * 
+ * @author <a href="mailto:duncan.doyle@redhat.com">Duncan Doyle</a>
+ */
+public interface QueryStrategy {
```

Comment:
It can't be handled by JPA because we don't use JPA. We use the jBPM AdvancedQueries API which does a direct SQL call into the DB. The reason we use the AdvancedQueries API instead of JPA is because it allows us to almost fully reuse the existing (remote) query interface and query services without needing to come up with all sorts of JPA-based queries. So main reason was that this is simply a small layer on top of Advanced Queries rather than something completely new.

The Advanced Queries API requires us to pre-register a query (which can be DB specific) and the mapping of columns to types (which can be DB specific), hence the decision to use a Strategy per DB type.